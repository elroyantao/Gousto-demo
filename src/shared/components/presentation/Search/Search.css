.Search {
  padding: 20px;
}
.Search-input {
  outline: none;
  font-size: 16px;
  border: none;
  border-bottom: 1px solid #645B57;
  color: #645B57;
}

.Search-input::placeholder {
  opacity: .7;
}
