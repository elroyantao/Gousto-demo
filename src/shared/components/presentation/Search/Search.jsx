import React, { Component } from 'react'
import PropTypes from 'prop-types'

import styles from './Search.css'

export default class Search extends Component {
  static propTypes = {
    searchTerm: PropTypes.string.isRequired,
    onSearch: PropTypes.func.isRequired
  }

  static defaultProps = {
    searchTerm: ''
  }

  handleChange = (event) => {
    this.props.onSearch(event.target.value)
  }
  render() {
    const { searchTerm } = this.props
    return (
      <div className={styles.Search}>
        <input
          type="search"
          className={styles['Search-input']}
          value={searchTerm}
          onChange={this.handleChange}
          placeholder="search product"
        />
      </div>
    )
  }
}
