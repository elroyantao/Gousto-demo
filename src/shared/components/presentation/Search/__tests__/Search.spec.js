import testComponentHelper from '../../../../lib/test-helper'
import Search from '../Search'

describe('<Search/>', () => {
  const renderComponent = testComponentHelper(Search)
  const initialProps = {
    searchTerm: 'fake term',
    onSearch: jest.fn()
  }

  describe('@renders', () => {
    it('in default state', () => {
      expect(renderComponent(initialProps).getTree())
        .toMatchSnapshot()
    })
    it('with empty searchTerm', () => {
      expect(renderComponent({
        ...initialProps,
        searchTerm: ''
      }).getTree())
        .toMatchSnapshot()
    })
  })

  describe('@events', () => {
    describe('onChange imput', () => {
      it('should call onSearch', () => {
        const changeEvent = { target: { value: 'changed search' } }
        const { wrapper, instance } = renderComponent(initialProps)
        expect(instance.props.onSearch).not.toHaveBeenCalled()
        wrapper.find('input').simulate('change', changeEvent)
        expect(instance.props.onSearch).toHaveBeenCalledTimes(1)
        expect(instance.props.onSearch).toHaveBeenLastCalledWith(changeEvent.target.value)
      })
    })
  })
})
