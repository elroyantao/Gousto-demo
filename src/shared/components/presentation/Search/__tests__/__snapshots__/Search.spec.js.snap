// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search/> @renders in default state 1`] = `
<div
  className="Search"
>
  <input
    className="Search-input"
    onChange={[Function]}
    placeholder="search product"
    type="search"
    value="fake term"
  />
</div>
`;

exports[`<Search/> @renders with empty searchTerm 1`] = `
<div
  className="Search"
>
  <input
    className="Search-input"
    onChange={[Function]}
    placeholder="search product"
    type="search"
    value=""
  />
</div>
`;
